iiod-client.c

```diff
@@ -1685,10 +1685,8 @@ int iiod_client_dequeue_block(struct iio_block_pdata *block, bool nonblock)
 
 	iio_mutex_lock(block->lock);
 
-	if (!block->enqueued) {
-		ret = -EBADF;
+	if (!block->enqueued)
```

Comment:
I agree is not an error but maybe something that apps should not do and try to fix... Just wondering if a warning or even debug message would make sense?

Reply:
So I'm just trying to make the behaviour consistent across the library. If we consider it is an error and not a no-op to dequeue an already dequeued block, then I'll have to change other parts of the library, because right now it's inconsistent.

I am fine with both approaches. But I'd prefer to return an error if it's something that the apps should not do.